warehouse/admin/static/js/warehouse.js

```diff
@@ -185,3 +185,42 @@ if (malwareReportsTable.length) {
   new $.fn.dataTable.Buttons(table, {buttons: ["copy", "csv", "colvis"]});
   table.buttons().container().appendTo($(".col-md-6:eq(0)", table.table().container()));
 }
+
+// Link Checking
+var links = document.querySelectorAll("a");
+links.forEach(function(link){
+  if (link.dataset.checkLinkUrl !== undefined) {
+    var reportLine = {bareUrl: link.href, url: link.dataset.checkLinkUrl, status:0, element : link};
+    fetch(reportLine.url, {
+      method: "GET",
+      mode: "cors",
+    })
+      .then(function(response) {
+        var responseText = "";
+        response.text().then((text) => {
+          responseText = text;
+          console.log(response.status, responseText);
+          if (response.status == 400 && responseText == "Unsupported content-type returned\n") {
+            reportLine.element.firstChild.classList.remove("fa-question");
+            reportLine.element.firstChild.classList.add("fa-check");
+            reportLine.element.firstChild.classList.add("text-green");
+            reportLine.status = 1;
+          } else {
+            reportLine.status = 0;
+            reportLine.element.firstChild.classList.remove("fa-question");
+            reportLine.element.firstChild.classList.add("fa-times");
+            reportLine.element.firstChild.classList.add("text-red");
+          }
+          console.log(reportLine);
+        });
+      })
+      .catch(function(error){
```

Comment:
Dunno why linter doesn't catch this?
```suggestion
      .catch(function(error) {
```

Reply:
41e60672e